### PR TITLE
Feature/external api

### PR DESCRIPTION
**Pull request by julianvasquez-wolox**
## Summary
- OpenLibraryService created.
- Consume of external api via RestTemplate.
- external api response map to Book object added to OpenLibraryService.
- findByISBN method added to BookController.

## Trello Card
- https://trello.com/c/sssae8Hk/22-integraci%C3%B3n-api-externa

### CODE REVIEW

File: src/main/java/wolox/training/service/OpenLibraryService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  private RestTemplate restTemplate = new RestTemplate();
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  public Optional<Book> bookInfo(String isbn) throws JsonProcessingException {
+    JsonNode jsonBook = getBookFromExternalAPI(isbn);
+    if(jsonBook.isEmpty()){
+      return Optional.empty();
+    }
+    Book book = new Book();
```
Comment (facucompe) [inline src/main/java/wolox/training/service/OpenLibraryService.java:28]: If we could have lombok builders i would tell you to use them here

File: src/main/java/wolox/training/service/OpenLibraryService.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  private String base_uri;
+
+  private RestTemplate restTemplate = new RestTemplate();
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  public Optional<Book> bookInfo(String isbn) throws JsonProcessingException {
+    JsonNode jsonBook = getBookFromExternalAPI(isbn);
+    if(jsonBook.isEmpty()){
+      return Optional.empty();
+    }
```
Comment (facucompe) [inline src/main/java/wolox/training/service/OpenLibraryService.java:27]: `return jsonBook.isEmpty() ? Optional.empty() : Optional.of(parseBookResponse(jsonBook));`

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -3,7 +3,8 @@ spring.datasource.username= postgres
 spring.datasource.password= postgres
 
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.PostgreSQLDialect
-
 spring.jpa.hibernate.ddl-auto = update
 
 server.port=8081
+
+openlibrary.url=https://openlibrary.org
```
Comment (facucompe) [inline src/main/resources/application.properties:10]:
```suggestion
openlibrary.url=https://openlibrary.org

```

**Review by facucompe — APPROVED** [review]
Some suggestions. No need to correct them

File: src/main/java/wolox/training/service/OpenLibraryService.java (hunk 2 above)
Comment (facucompe) [reply src/main/java/wolox/training/service/OpenLibraryService.java:27]: `parseBookResponse` would have all the logic below the if

**Review by rabinovichm — APPROVED** [review]
Nice work 👏

**Pull request merged**